src/main/scala/ai/privado/languageEngine/java/tagger/collection/GrpcCollectionTagger.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package ai.privado.languageEngine.java.tagger.collection
+
+import ai.privado.model.{InternalTag}
+import ai.privado.utility.Utilities._
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.passes.CpgPass
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+class GrpcCollectionTagger(cpg: Cpg) extends CpgPass(cpg) {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def run(builder: DiffGraphBuilder): Unit = {
+
+    val grpcCollectionPoints = cpg.method
+      .where(_.call.name("onCompleted")) // grpc server always call this in handler when grpc-java is used
```

Comment:
I think we can qualify this `onCompleted` method further by adding number of params and `StreamObserver` in methodFullName filter

`val OnCompletedPattern = "(?i)(.*)(StreamObserver)(.*)(onCompleted)(.*)([(](0)[)])$"`

Reply:
Maybe instead of using regex for param count, we can check via argument index on the call. Let me see if it's available.